Sources/BBCodeParser.php

```diff
@@ -4099,7 +4102,7 @@ protected function transformToHtml(array $tag, array $params): void
 				return;
 			}
 
-			$data = substr($this->message, $this->pos1, $pos2 - $this->pos1);
+			$data = substr($this->message, $this->pos1, (int) ($pos2 - $this->pos1));
```

Comment:
Casting the result of the expression to int is unnecessary. The result of the arithmetic operation will already be an integer.